src/util/simplify_expr.cpp

```diff
@@ -1826,6 +1827,44 @@ optionalt<std::string> simplify_exprt::expr2bits(
   return {};
 }
 
+const array_exprt *simplify_exprt::get_content(const refined_string_exprt &s)
```

Comment:
`try_get_string_data_array` perhaps? Also suggest a brief comment like
``/// If `s.content()` is of the form `&some_symbol[e]`, where `some_symbol` is an array-typed symbol and `e` is any expression, return `some_symbol`; otherwise return null.``